src/components/TrapFocus/TrapFocus.tsx

```diff
@@ -15,81 +18,98 @@ export interface TrapFocusProps {
   children?: React.ReactNode;
 }
 
-interface State {
-  shouldFocusSelf: boolean | undefined;
-}
-
-export class TrapFocus extends React.PureComponent<TrapFocusProps, State> {
-  state: State = {
-    shouldFocusSelf: undefined,
-  };
+export function TrapFocus({trapping = true, children}: TrapFocusProps) {
+  const [shouldFocusSelf, setFocusSelf] = useState<boolean | undefined>(
+    undefined,
+  );
 
-  private focusTrapWrapper: HTMLElement;
+  const focusTrapWrapper = useRef<HTMLDivElement>(null);
 
-  componentDidMount() {
-    this.setState(this.handleTrappingChange());
-  }
-
-  handleTrappingChange() {
-    const {trapping = true} = this.props;
-
-    if (this.focusTrapWrapper.contains(document.activeElement)) {
-      return {shouldFocusSelf: false};
+  const handleTrappingChange = () => {
+    if (
+      focusTrapWrapper.current &&
+      focusTrapWrapper.current.contains(document.activeElement)
+    ) {
+      return false;
     }
+    return trapping;
+  };
 
-    return {shouldFocusSelf: trapping};
-  }
-
-  render() {
-    const {children} = this.props;
-
-    return (
-      <Focus disabled={this.shouldDisable} root={this.focusTrapWrapper}>
-        <div ref={this.setFocusTrapWrapper}>
-          <EventListener event="focusout" handler={this.handleBlur} />
-          {children}
-        </div>
-      </Focus>
-    );
-  }
-
-  private get shouldDisable() {
-    const {trapping = true} = this.props;
-    const {shouldFocusSelf} = this.state;
+  useComponentDidMount(() => setFocusSelf(handleTrappingChange()));
 
+  const shouldDisable = () => {
     if (shouldFocusSelf === undefined) {
       return true;
     }
 
     return shouldFocusSelf ? !trapping : !shouldFocusSelf;
-  }
+  };
 
-  private setFocusTrapWrapper = (node: HTMLDivElement) => {
-    this.focusTrapWrapper = node;
+  const handleKeyDownLastElement = (event: any) => {
+    if (event.keyCode === Key.Tab && focusTrapWrapper.current) {
+      !event.shiftKey && focusFirstFocusableNode(focusTrapWrapper.current);
+
+      document.removeEventListener('keydown', handleKeyDownLastElement);
+    }
+  };
+
+  const handleKeyDownFirstElement = (event: any) => {
+    if (event.keyCode === Key.Tab && focusTrapWrapper.current) {
+      event.shiftKey && focusFirstFocusableNode(focusTrapWrapper.current);
```

Comment:
This should really be `event.shiftKey && focusLastFocusableNode(focusTrapWrapper.current)`. Yet this appears to "skip" the last element that we should be focusing.

It appears that we do focus the last element, and then <kbd>tab</kbd> + <kbd>shift</kbd> is also taken into consideration and back tabs a second time.

A work around for this is to focus the first element, and assuming that <kbd>tab</kbd> + <kbd>shift</kbd> is back tabbing on its own, focuses the last element.